stat/roc.go

```diff
@@ -127,3 +129,68 @@ func ROC(cutoffs, y []float64, classes []bool, weights []float64) (tpr, fpr, thr
 
 	return tpr, fpr, cutoffs
 }
+
+// TOC returns the Total Operating Characteristic for the classes provided
+// and the minimum and maximum bounds for the TOC.
+//
+// The input y values that correspond to classes and weights must be sorted
+// in ascending order. classes[i] is the class of value y[i] and weights[i]
+// is the weight of y[i]. SortWeightedLabeled can be used to sort classes
+// together with weights by the rank variable.
+//
+// The returned ntp values can be interpreted as the number of true positives
+// where values above the given rank are assigned class true for each given
+// rank from 1 to len(classes). The values of min and max provide the minimum
+// and maximum possible number of true values for the set of classes.
+//
+// If weights is nil, all weights are treated as 1. If weights is not nil
+// it must have the same length as classes, otherwise TOC will panic.
```

Comment:
I would add here `When weights are not nil, the calculation of min and max allows for partial assignment of single data points`. To explain in more detail: for example if the total weight of the true values is 8, and the first three values have weights [3 3 3] the third (well fourth) max value will be 8, i.e. min(8, 3 + 3 + 3), even though there is no way to actually make 8 from any combination of [3 3 3]. I agree this is the correct implementation to use, primarily because computing the actual max in the weighted case is combinatorially hard, but I think we should note that we are using this simplification.

Reply:
Done